example/src/main/java/io/netty/example/tunechoserver/TunEchoServer.java

```diff
@@ -0,0 +1,452 @@
+                (inPacket.protocol() == TunTapPacket.PROTOCOL_IPV6 && ipHeader.ipVersion != InetUtils.IP_VERSION_6)) {
+                return;
+            }
+
+            // Create a copy of the inbound packet that will become the response packet.
+            outPacketBuf = inPacketBuf.copy();
+            outPacket = new TunTapPacket(inPacket.protocol(), outPacketBuf);
+
+            // Swap the source and destination addresses in the response packet.
+            swapSourceDestAddresses(outPacketBuf, ipHeader);
+
+            // If the packet is an ICMP or ICMPv6 packet...
+            if ((ipHeader.ipVersion == InetUtils.IP_VERSION_4 && ipHeader.protocol == InetUtils.IP_PROTOCOL_ICMP) ||
+                (ipHeader.ipVersion == InetUtils.IP_VERSION_6 && ipHeader.protocol == InetUtils.IP_PROTOCOL_ICMPV6)) {
+
+                // Decode ICMP header.
+                InetUtils.ICMPHeader icmpHeader = InetUtils.decodeICMPHeader(inPacketBuf, ipHeader.headerLength);
+
+                // Ignore anything other than an ICMP or ICMPv6 Echo Request
+                if ((ipHeader.ipVersion == InetUtils.IP_VERSION_4 &&
+                     icmpHeader.type != InetUtils.ICMP_MESSAGE_TYPE_ECHO_REQUEST) ||
+                    (ipHeader.ipVersion == InetUtils.IP_VERSION_6 &&
+                     icmpHeader.type != InetUtils.ICMPV6_MESSAGE_TYPE_ECHO_REQUEST) ||
+                    icmpHeader.code != 0) {
+                    return;
+                }
+
+                // Change the message type to Echo Reply.
+                int newType = (ipHeader.ipVersion == InetUtils.IP_VERSION_4)
+                    ? InetUtils.ICMP_MESSAGE_TYPE_ECHO_REPLY
+                    : InetUtils.ICMPV6_MESSAGE_TYPE_ECHO_REPLY;
+                changeICMPTypeAndCode(outPacketBuf, ipHeader, icmpHeader, newType, 0);
+
+            // Otherwise if the packet is a UDP packet...
+            } else if (ipHeader.protocol == InetUtils.IP_PROTOCOL_UDP) {
+
+                // Decode the UDP header.
+                InetUtils.UDPHeader udpHeader = InetUtils.decodeUDPHeader(inPacketBuf, ipHeader.headerLength);
+
+                // Swap the source and destination port numbers in the response packet.
+                swapSourceDestPorts(outPacketBuf, ipHeader, udpHeader);
+
+            // Otherwise, not an ICMP or UDP packet, so ignore it.
+            } else {
+                return;
+            }
+
+            // Write the response packet to the TUN device.
+            channel.writeAndFlush(outPacket);
+            outPacket = null;
+            outPacketBuf = null;
+        } finally {
+            if (outPacket != null) {
+                outPacket.release();
+            } else if (outPacketBuf != null) {
+                outPacketBuf.release();
+            }
+        }
+
+        // If requested, provoke GC as an aid to detecting buffer leaks.
+        if (_forceGC) {
+            System.gc();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Don't try to be smarter than the JVM, remove this call to run the garbage collector. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1215)